### DISCUSSION TITLE
Handing Sub-Resource Queries

### DISCUSSION THREAD

**Discussion by ICarpenter** [TanStack/query#8291] [Q&A]
I'm starting a new frontend project and am leaning towards using the TanStack ecosystem. One challenge I've run into is how to handle API calls that return sub-resources (for example, a chat with messages). I initially solved this using `QueryObserver` and `queryClient.setQueryData`, but this approach feels a bit verbose. Sub-resources, especially in GraphQL, are quite common, so I'm wondering if others have encountered this issue and found a more elegant solution. Any advice or feedback would be greatly appreciated!

**Comment by TkDodo** [comment]
see: https://tkdodo.eu/blog/seeding-the-query-cache#seeding-details-from-lists
